src/python/bindings/bind.cpp

```diff
@@ -83,5 +84,30 @@ PYBIND11_MODULE(smartredisPy, m) {
         .def("get_meta_scalars", &PyDataset::get_meta_scalars)
         .def("get_meta_strings", &PyDataset::get_meta_strings)
         .def("get_name", &PyDataset::get_name);
+
+    // Register exception classes
+    py::register_exception_translator([](std::exception_ptr p) {
+        try {
+            if (p) std::rethrow_exception(p);
+        }
+        catch (const _smart_bad_alloc& e) {
+            PyErr_SetString(PyExc_MemoryError, e.what());
+        }
+        catch (const _smart_database_error& e) {
```

Comment:
I don't know if this maps to an ``OSError``, see overview comments.

Reply:
This is a placeholder mapping until a future time when we have custom error classes in Python that things can more cleanly map to

Reply:
Deferred to Python error handling ticket